Завершаем вёрстку всех компонентов

TimurPokatov opened the pull request:
---
:mortar_board: [Завершаем вёрстку всех компонентов](https://up.htmlacademy.ru/htmlcss/35/user/2196841/tasks/15)

:boom: https://htmlacademy-htmlcss.github.io/2196841-sedona-35/9/

TimurPokatov committed:
выполняет задание 9

andreysgra commented on catalog.html:
```diff
@@ -174,12 +172,12 @@ <h3 class="hotel-card-title">GreenTree Inn</h3>
-            <input class="field" placeholder="Ваш e-mail" type="email" name="subscribe-email" id="subscribe-email" required>
-            <button class="subscribe-button" type="submit">Подписаться</button>
+            <input class="field-subscribe field-background" placeholder="Ваш e-mail" type="email" name="subscribe-email" id="subscribe-email" required>
+            <button class="subscribe-button cyan-button" type="submit"><span class="subscribe-button-text">Подписаться</span></button>
```
`span` то зачем здесь? В кнопке есть явный текст, поэтому просто
```suggestion
            <button class="subscribe-button cyan-button" type="submit">Подписаться</button>
```

andreysgra commented on styles/styles.css:
```diff
@@ -257,19 +232,44 @@ body {
   color: #ffffff;
+  font-weight: 700;
+  text-transform: uppercase;
+  transform: scaleX(0.85);
```
Это свойство не нужно. А не совпадает текст с макетом потому что Arial, а не PT Sans. Нужно явно указать в свойстве `font-family`. Это свойство в кнопках не наследуется.
![image](https://user-images.githubusercontent.com/17790771/201536490-6dbcad3c-1833-43eb-b118-3274023d838e.png)
Также нужно задать размер шрифта и высоту строки.
![image](https://user-images.githubusercontent.com/17790771/201536566-27ce243a-63ab-4f12-9bfa-4abb243b638b.png)

andreysgra commented on index.html:
```diff
@@ -105,19 +103,19 @@ <h3 class="tips-title">Сувениры</h3>
-            <button class="subscribe-button" type="submit">Подписаться</button>
+            <input class="field-subscribe" placeholder="Ваш e-mail" type="email" name="subscribe-email" id="subscribe-email" required>
+            <button class="subscribe-button cyan-button" type="submit"><span
+                class="subscribe-button-text">Подписаться</span></button>
```
```suggestion
            <button class="subscribe-button cyan-button" type="submit">Подписаться</button>
```

andreysgra commented on index.html:
```diff
@@ -157,11 +167,66 @@ <h2 class="visually-hidden">Звоните нам!</h2>
+              <label class="modal-text" for="adults-number">
+                Взрослые:
+              </label>
+              <button class="field-button field-button-minus field-background modal-button-cursor"><span class="visually-hidden">Уменьшить число.</span></button>
```
Для этой и других кнопок "-" и "+" тип не указан.

andreysgra commented on index.html:
```diff
@@ -157,11 +167,66 @@ <h2 class="visually-hidden">Звоните нам!</h2>
+                <label class="modal-text-inner" for="children-number">
+                  Дети:
+                </label>
+                <button class="tooltip-button modal-round-button modal-button-cursor" type="button"><span class="visually-hidden">Тултип.</span></button>
```
А само окно тултипа где?

andreysgra commented on styles/styles.css:
```diff
@@ -842,3 +878,230 @@ body {
+  background-color: #e6e6e6;
+}
+
+.modal-field-text:focus {
+  background-color: #e6e6e6;
+  border: 3px solid #83b3d3;
+}
```
1. Во всех полях ввода этой формы фокус приводит к изменению размеров окружения.
Состояние покоя:
![image](https://user-images.githubusercontent.com/17790771/201536962-93d92737-91ba-4379-9eb5-88cfcf182de5.png)
Фокус в первом поле ввода:
![image](https://user-images.githubusercontent.com/17790771/201537007-b95027dd-38e4-4488-8fbe-cff660394314.png)

Суть в том, что свойство `border` влияет на геометрию элемента. Вместо него используйте свойство `outline`.

2. При навигации клавиатурой или при клике мыши в поле ввода наблюдаем ещё и чёрную рамку:
![image](https://user-images.githubusercontent.com/17790771/201537103-4db33b17-2717-4c6d-a2e0-a0197771366f.png)
Это браузерные стили для фокуса. Дело в том, что современные браузеры отлавливают это событие по псевдо классу `:focus-visible`. Поэтому лучше писать стили не для `:focus`, а для `:focus-visible`.

TimurPokatov committed:
исправляет ошибки в 9 задании

Всё, кроме тултипа - реально не успел. Но надеюсь выложить его в 10 пулреквесте

TimurPokatov committed:
исправляет ошибки и недочеты в 9 задании
